File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        lowestValue = prices[0]
        maxProfit = 0

        for i in prices:
            if i < lowestValue:
                lowestValue = i

            if i - lowestValue > maxProfit:
                maxProfit = i - lowestValue
            
        return(maxProfit)

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices)
    {
        int maxProfit = 0;
        for(int sell = 1, buy = 0; sell < prices.Length; sell++)
        {
            if(prices[buy] < prices[sell])
            {
               maxProfit = prices[sell] - prices[buy] > maxProfit ? prices[sell] - prices[buy] : maxProfit;  
            }
            else
            {
                buy = sell;
            }
        }
        return maxProfit;
    }
}